src/containerDeployment/dockerUtils.ts

```diff
@@ -0,0 +1,688 @@
+            stepAction: checkDockerInstallation,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: [],
+            headerText: ContainerDeployment.startDockerHeader,
+            bodyText: ContainerDeployment.startDockerBody,
+            errorLink: "https://docs.docker.com/engine/",
+            errorLinkText: ContainerDeployment.startDockerEngine,
+            stepAction: startDocker,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: [],
+            headerText: ContainerDeployment.startDockerEngineHeader,
+            bodyText: ContainerDeployment.startDockerEngineBody,
+            stepAction: checkEngine,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: ["containerName", "password", "version", "hostname", "port"],
+            headerText: ContainerDeployment.creatingContainerHeader,
+            bodyText: ContainerDeployment.creatingContainerBody,
+            stepAction: startSqlServerDockerContainer,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: ["containerName"],
+            headerText: ContainerDeployment.settingUpContainerHeader,
+            bodyText: ContainerDeployment.settingUpContainerBody,
+            stepAction: checkIfContainerIsReadyForConnections,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: [],
+            headerText: ContainerDeployment.connectingToContainerHeader,
+            bodyText: ContainerDeployment.connectingToContainerBody,
+            stepAction: undefined,
+        },
+    ];
+}
+
+/**
+ * Handles the result of a Docker command and updates the corresponding step statuses accordingly.
+ */
+export function setStepStatusesFromResult(
+    result: DockerCommandParams,
+    currentStep: DockerStepOrder,
+    steps: DockerStep[],
+): DockerStep[] {
+    if (result.success) {
+        steps[currentStep].loadState = ApiStatus.Loaded;
+    } else {
+        steps[currentStep].loadState = ApiStatus.Error;
+        steps[currentStep].errorMessage = result.error;
+        steps[currentStep].fullErrorText = truncateAndSanitizeErrorTextIfNeeded(
+            result.fullErrorText,
+        );
+        for (let i = currentStep + 1; i < steps.length; i++) {
+            steps[i].loadState = ApiStatus.Error;
+            steps[i].errorMessage = ContainerDeployment.previousStepFailed;
+        }
+    }
+    return steps;
+}
+
+/**
+ * Truncates the error text if it exceeds the maximum length and sanitizes sensitive info.
+ */
+export function truncateAndSanitizeErrorTextIfNeeded(errorText: string): string {
+    // Sanitize
+    const sanitized = errorText.replace(/(SA_PASSWORD=)([^ \n]+)/gi, '$1******"');
+
+    // Truncate if necessary
+    if (sanitized.length > MAX_ERROR_TEXT_LENGTH) {
+        return `${sanitized.substring(0, MAX_ERROR_TEXT_LENGTH)}...`;
+    }
+
+    return sanitized;
+}
+
+/**

```

Comment:
NIT: add link to the password policy.

Reply:
https://learn.microsoft.com/en-us/sql/linux/sql-server-linux-docker-container-configure?view=sql-server-ver17&pivots=cs1-bash#mount-a-host-directory-as-data-volume